File: section29/OrderManager/OrderManager.Core/DTOs/OrderResponse.cs
using OrderManager.Core.Domain.Entities;

namespace OrderManager.Core.DTOs;

public class OrderResponse
{
    public Guid OrderId { get; set; }
    public string? OrderNumber { get; set; }
    public string? CustomerName { get; set; }
    public DateTime? OrderDate { get; set; }
    public decimal TotalAmount { get; set; }

    public OrderResponse(Order order)
    {
        OrderId = order.OrderId;
        OrderNumber = order.OrderNumber;
        CustomerName = order.CustomerName;
        OrderDate = order.OrderDate;
        TotalAmount = order.TotalAmount;
    }
}

File: section29/OrderManager/OrderManager.API/ClientApp/OrdersAngularApp/src/app/models/order-response.ts
export class OrderResponse {
  orderId: string;
  orderNumber: string;
  customerName: string;
  orderDate: Date;
  totalAmount: number | null;

  constructor(orderId: string,
    orderNumber: string,
    customerName: string,
    orderDate: Date,
    totalAmount: number | null) {
    this.orderId = orderId;
    this.orderNumber = orderNumber;
    this.customerName = customerName;
    this.orderDate = orderDate;
    this.totalAmount = totalAmount;
  }
}
